src/core/smtp/MCSMTPSession.cpp

```diff
@@ -646,15 +683,25 @@ void SMTPSession::sendMessage(Address * from, Array * recipients, Data * message
     }
     MCLog("send");
     if ((mSmtp->esmtp & MAILSMTP_ESMTP_PIPELINING) != 0) {
-        r = mailesmtp_send_quit(mSmtp, MCUTF8(from->mailbox()), 0, NULL,
+        r = mailsmtp_data_message_quit_no_disconnect(mSmtp, MCUTF8(from->mailbox()), 0, NULL,
```

Comment:
You probably wanted to use that function instead?
mailesmtp_send_quit_no_disconnect()